Detour/Include/DetourCommon.h

```diff
@@ -482,6 +483,15 @@ inline void dtSwapEndian(float* v)
 void dtRandomPointInConvexPoly(const float* pts, const int npts, float* areas,
 							   const float s, const float t, float* out);
 
+template<typename TypeToRetrieveAs, typename BufferType>
+TypeToRetrieveAs* getThenAdvanceBufferPointer(BufferType*& buffer, const size_t distanceToAdvance)
```

Comment:
It could, yes - I thought I saw places using `signed char*` which would also be valid but after a second look you're right, it's only ever `unsigned char*`. I'll remove the template argument for simplicity.